### PR TITLE
cmd: split flux-job into separate source files

### PR DESCRIPTION
**Pull request by grondo**
This PR splits `flux-job.c` into separate source files (roughly per-subcommand). The source was getting a bit unmanageable with everything in one big file.

This is just code reorganization and shouldn't represent any other change.

### CODE REVIEW

File: src/cmd/job/last.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* flux-job id */
```
Comment (chu11) [inline src/cmd/job/last.c:11]: s/id/last/

File: src/cmd/job/urgency.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* flux-job id */
```
Comment (chu11) [inline src/cmd/job/urgency.c:11]: s/id/urgency/

File: src/cmd/job/timeleft.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* flux-job id */
```
Comment (chu11) [inline src/cmd/job/timeleft.c:11]: s/id/timeleft/

File: src/cmd/job/taskmap.c [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* flux-job id */
```
Comment (chu11) [inline src/cmd/job/taskmap.c:11]: s/id/taskmap/

**Review by chu11 — APPROVED** [review]
LGTM, noticed a few nits

File: src/cmd/job/taskmap.c (hunk 4 above)
Comment (grondo) [reply src/cmd/job/taskmap.c:11]: D'oh! Good catch on those! :facepalm:

**Comment by garlick** [comment]
Nice cleanup, thanks for doing that!  It was getting out of hand.

**Comment by grondo** [comment]
Rebased on latest changes to `flux-job.c` and force-pushed.

**Comment by grondo** [comment]
Looks like tests are passing so I'll set MWP. Thanks!

**Pull request merged**